flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetSort.scala

```diff
@@ -71,6 +72,21 @@ class DataSetSort(
     )
   }
 
+  override def estimateRowCount(metadata: RelMetadataQuery): Double = {
+    val inputRowCnt = metadata.getRowCount(this.getInput)
+    if (inputRowCnt == null) {
+      inputRowCnt
+    } else {
+      val rowCount = Math.max(inputRowCnt - limitStart, 0D)
```

Comment:
Hmm, the estimation 0 only  happens if inputRowCount <= start and (fetch  is null or fetchValue<=0). I think this estimation is reasonable in this case.  Besides, why choose return 1 instead of 0.1 or 0.01 or other values?